### PR TITLE
Cache release-manager CLI

### PR DESCRIPTION
**Pull request by jbudz**
The release manager CLI is roughly ~350mb.  We expect this to run on
every commit when producing artifacts.  Instead of pulling this at
runtime, lets cache the image to see if we can limit networking errors.

This isn't ready to merge, need to figure out how to test this.  @brianseeders do you have any concerns with this going in the base agent image?

### CODE REVIEW

**Comment by jbudz** [comment]
@brianseeders I'm not really sure if this is the best approach, any thoughts here?

File: agents/ubuntu/ansible/roles/common/tasks/misc.yml [hunk 1]
```diff
@@ -77,3 +77,19 @@
     content: "{{ lookup('hashi_vault', 'secret=secret/kibana-issues/dev/buildkite-vault-credentials:secret_id') }}"
     mode: "0600"
     owner: "buildkite-agent"
+
+- name: Log into docker.elastic.co
+  docker_login:
+    registry: docker.elastic.co
+    username: "{{ lookup('hashi_vault', 'secret= secret/kibana-issues/dev/container-registry:username') }}"
```
Comment (brianseeders) [inline agents/ubuntu/ansible/roles/common/tasks/misc.yml:84]: Looks like you need to remove the space after `secret= ` in all of these

```
fatal: [127.0.0.1]: FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'hashi_vault'. Error was a <class 'ansible.errors.AnsibleError'>, original message: hashi_vault lookup plugin needs key=value pairs, but received secret= secret/kibana-issues/dev/container-registry:username"
}
```

**Comment by brianseeders** [comment]
Can you add `docker` to `requirements.txt`?

```
 "msg": "Failed to import the required Python library (Docker SDK for Python: docker above 5.0.0 (Python >= 3.6) or docker before 5.0.0 (Python 2.7) or docker-py (Python 2.6)) on a20976016d23's Python /usr/bin/python3. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter, for example via `pip install docker` (Python >= 3.6) or `pip install docker==4.4.4` (Python 2.7) or `pip install docker-py` (Python 2.6). The error was: No module named 'docker'"
```

File: agents/ubuntu/ansible/roles/common/tasks/misc.yml (hunk 1 above)
Comment (jbudz) [reply agents/ubuntu/ansible/roles/common/tasks/misc.yml]: :+1: c7ffb710261a3341ab7323b22f09ed7016d23491

**Comment by jbudz** [comment]
> Can you add `docker` to `requirements.txt`?
> 
> ```
>  "msg": "Failed to import the required Python library (Docker SDK for Python: docker above 5.0.0 (Python >= 3.6) or docker before 5.0.0 (Python 2.7) or docker-py (Python 2.6)) on a20976016d23's Python /usr/bin/python3. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter, for example via `pip install docker` (Python >= 3.6) or `pip install docker==4.4.4` (Python 2.7) or `pip install docker-py` (Python 2.6). The error was: No module named 'docker'"
> ```

Pushed with 0833d8d5732129ff224b71e4d8286a2d1ffb3c2d

**Review by brianseeders — APPROVED** [review]

**Pull request merged**